### PR TITLE
fix flaky array query slt test

### PR DESCRIPTION
**Pull request by leoyvens**
This was failing for me locally.

### CODE REVIEW

File: datafusion/sqllogictest/test_files/array_query.slt [hunk 1]
```diff
@@ -93,11 +93,11 @@ query ??I
 SELECT * FROM data WHERE null = column2;
 ----
 
-query ??I
+query ??I rowsort
```
Comment (comphead) [inline datafusion/sqllogictest/test_files/array_query.slt:96]: wondering should we also fix the test below?

Comment (leoyvens) [reply datafusion/sqllogictest/test_files/array_query.slt:96]: The one right below has a single output row, so it doesn't need to be sorted.

**Review by alamb — APPROVED** [review]
Thank you @leoyvens  and @comphead 

I believe this test was newly introduced in https://github.com/apache/datafusion/pull/11091 from @jayzhan211 today so 👍

**Comment by alamb** [comment]
Merging to keep CI stable

**Pull request merged**